13-multiplecomponents-lab/src/03-shoppinglist-filters.jsx

```diff
@@ -1,7 +1,170 @@
-/**
- * Lab 03: Shopping List Filters
- *
- * Add two filters to the shopping list built in previous lab:
- *  - Ability to show only products matching a given name
- *  - Ability to show only products that were fully bought
- */
+/**
+ * Lab 03: Shopping List Filters
+ *
+ * Add two filters to the shopping list built in previous lab:
+ *  - Ability to show only products matching a given name
+ *  - Ability to show only products that were fully bought
+ */
+import React from "react";
+
+function ShoppingItemsCount(props) {
+	return <p>Total items: {props.count}</p>
+}
+
+export default class ShoppingListFilter extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			items: [],
+		};
+		this.getParameters.bind(this);
+	}
+
+	addItem() {
+		const nameInput = this.refs.newItemName;
+		const imageInput = this.refs.newItemImage;
+		const neededQuantityInput = this.refs.newItemNeededQuantity;
+		const id = new Date().getTime();
+		const items = this.state.items;
+		items.push({id: id, name: nameInput.value, image: imageInput.value, neededQuantity: neededQuantityInput.value});
+		this.setState(items);
+
+		nameInput.value = '';

```

Comment:
I strongly advice against changing input values or html in general directly. For that you have React. So I advice to add to all inputs onChange handler. Make it one abstract handler for all inputs which will get state parameter name through partial application.

Something like that:

```
<input type="text" value={ this.state.newItem.name } onChange={ this.handleNewItemChange.bind(this,"name")} />

```

You will also have another state variable newItem to store all state changes.
When adding newItem you can simply reset your newItem, and your inputs will be rerendered. To make it even better, extract your input with onChange handler to different component and reuse it.